plugins/package-managers/maven/src/main/kotlin/utils/DependencyTreeLogger.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2025 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packagemanagers.maven.utils
+
+import org.codehaus.plexus.logging.AbstractLogger
+
+/**
+ * A special logger implementation that supports finding the output of the Maven dependency tree plugin.
+ *
+ * When running a Maven build programmatically, the generated log output is affected by the logging configuration of
+ * the client application. This also affects the log level and the log format. Because of this, it can be problematic
+ * to detect the output of the Maven dependency tree plugin in the log output. This logger implementation solves this
+ * problem by using a deterministic log format for all output generated by the plugin.
+ */
+internal class DependencyTreeLogger : AbstractLogger(LEVEL_DEBUG, "DependencyTreeLogger") {
+    companion object {
+        /** The name of the logger used by the Maven dependency tree plugin. */
+        const val DEPENDENCY_TREE_LOGGER = "org.apache.maven.plugins.dependency.tree.TreeMojo"
+
+        /**
+         * Generate a deterministic log output for the given log [message] from the dependency plugin.
+         */
+        fun formatMessage(message: String) = "[$DEPENDENCY_TREE_LOGGER] $message"
+
+        /**
+         * Return a flag whether the given log [message] is generated by the dependency tree plugin.
+         */
+        fun isDependencyTreeLog(message: String) = message.startsWith("[$DEPENDENCY_TREE_LOGGER]")
+    }
+
+    override fun getChildLogger(name: String?) = this
+
+    override fun debug(message: String, throwable: Throwable?) = log(message, throwable)
+
+    override fun error(message: String, throwable: Throwable?) = log(message, throwable)
+
+    override fun fatalError(message: String, throwable: Throwable?) = log(message, throwable)
+
+    override fun info(message: String, throwable: Throwable?) = log(message, throwable)
+
+    override fun warn(message: String, throwable: Throwable?) = log(message, throwable)
+
+    /**
+     * Log the given [message] and optional [throwable] in the standard format defined by this logger.
+     */
+    private fun log(message: String, throwable: Throwable?) {
+        println(formatMessage(message))
```

Comment:
I understand why you're using `println()`, but we have an unwritten policy of not using `println()` in library code. Can't you just forward to the regular logger?

Reply:
No, the whole point of this class is to be independent on the log format generated by the active logging system. But maybe it is an option to redirect the output to a file; this would also simplify the filtering of the output produced by the plugin.

Reply:
Yes, this approach worked. Together with using the JSON output option, parsing of dependency tree information could be simplified.
Note that there is an option of the Dependency Tree plugin to specify an [output file](https://maven.apache.org/plugins/maven-dependency-plugin/tree-mojo.html#outputfile). However, in my tests, this did not work for multi-module projects, since the file was overwritten for every project even when specifying the additional [appendOutput](https://maven.apache.org/plugins/maven-dependency-plugin/tree-mojo.html#appendoutput) option.